plugins/lang/python/python_loader.cc

```diff
@@ -497,6 +501,125 @@ runPythonScript(std::shared_ptr<PythonContext> titl,
     };
 }
 
+void
+PythonPlugin::
+injectMldbWrapper(PythonSubinterpreter & pyControl)
+{
+    std::string code = R"code(
+
+
+class mldb_wrapper(object):
+
+    @staticmethod
+    def wrap(mldb):
+        return Wrapped(mldb)
+
+    class ResponseException(Exception):
+        def __init__(self, response):
+            self.response = response
+
+        def __str__(self):
+            return ('Response status code: {r.status_code}. '
+                    'Response text: {r.text}'.format(r=self.response))
+
+    class Response(object):
+
+        import json as jsonlib
+
+        def __init__(self, url, raw_response):
+            self.url         = url
+            self.headers     = {k: v for k, v in
+                                raw_response.get('headers', {})}
+            self.status_code = raw_response['statusCode']
+            self.text        = raw_response.get('response', '')
+            self.raw         = raw_response
+
+            self.apparent_encoding = 'unimplemented'
+            self.close             = 'unimplemented'
+            self.conection         = 'unimplemented'
+            self.elapsed           = 'unimplemented'
+            self.encoding          = 'unimplemented'
+            self.history           = 'unimplemented'
+            self.iter_content      = 'unimplemented'
+            self.iter_lines        = 'unimplemented'
+            self.links             = 'unimplemented'
+            self.ok                = 'unimplemented'
+            self.raise_for_status  = 'unimplemented'
+            self.reason            = 'unimplemented'
+            self.request           = 'unimplemented'
+
+        def json(self):
+            return self.__class__.jsonlib.loads(self.text)
+
+    class wrap(object):
+        def __init__(self, mldb):
+            self._mldb = mldb
+            import functools
+            self.post = functools.partial(self._post_put, 'POST')
+            self.put = functools.partial(self._post_put, 'PUT')
+
+        def _perform(self, method, url, *args, **kwargs):
+            raw_res = self._mldb.perform(method, url, *args, **kwargs)
+            response = mldb_wrapper.Response(url, raw_res)
+            if response.status_code < 200 or response.status_code >= 400:
+                raise mldb_wrapper.ResponseException(response)
+            return response
+
+        def log(self, thing):
+            self._mldb.log(str(thing))
```

Comment:
can this pretty-print dictionaries? i.e. with indentation and without the `u'...'` that comes from the output of `json.loads()`?